daft/delta_lake/delta_lake_scan.py

```diff
@@ -0,0 +1,187 @@
+        partition_columns = set(self._table.metadata().partition_columns)
+        self._partition_keys = [
+            PartitionField(field._field) for field in self._schema if field.name in partition_columns
+        ]
+
+    def schema(self) -> Schema:
+        return self._schema
+
+    def display_name(self) -> str:
+        return f"DeltaLakeScanOperator({self._table.metadata().name})"
+
+    def partitioning_keys(self) -> list[PartitionField]:
+        return self._partition_keys
+
+    def multiline_display(self) -> list[str]:
+        return [
+            self.display_name(),
+            f"Schema = {self._schema}",
+            f"Partitioning keys = {self.partitioning_keys()}",
+            # TODO(Clark): Improve repr of storage config here.
+            f"Storage config = {self._storage_config}",
+        ]
+
+    def to_scan_tasks(self, pushdowns: Pushdowns) -> Iterator[ScanTask]:
+        # TODO(Clark): Push limit and filter expressions into deltalake action fetch, to prune the files returned.
+        paths = self._table.file_uris()
+        add_actions: pa.RecordBatch = self._table.get_add_actions()
+        assert len(paths) == add_actions.num_rows
+        if len(self.partitioning_keys()) > 0 and pushdowns.partition_filters is None:
+            logging.warn(
+                f"{self.display_name()} has partitioning keys = {self.partitioning_keys()}, but no partition filter was specified. This will result in a full table scan."
+            )
+        # NOTE: We don't assume that table.file_uris() and table.get_add_actions() return file entries in the same order, just to be safe.
+        # TODO(Clark): Keep action table in pyarrow form for the sake of efficiency and dtype handling?
+        actions: dict[str, dict] = {row["path"]: row for row in add_actions.to_pylist()}
+        del add_actions
+        limit_files = pushdowns.limit is not None and pushdowns.filters is None and pushdowns.partition_filters is None
+        rows_left = pushdowns.limit if pushdowns.limit is not None else 0
+        scan_tasks = []
+        for task_idx in range(len(paths)):
+            if limit_files and rows_left <= 0:
+                break
+
+            path = paths[task_idx]
+
+            # NOTE: We assume that the paths in the transaction log consist of only the file name.
+            file_name = os.path.split(path)[-1]
+            if file_name not in actions:
+                raise ValueError(f"File name {file_name} not in Delta Lake transaction log: {list(actions.keys())}")
+
+            row = actions[file_name]
+            record_count = row["num_records"]
+            try:
+                size_bytes = row["size_bytes"]
+            except KeyError:
+                size_bytes = None
+            file_format_config = FileFormatConfig.from_parquet_config(ParquetSourceConfig())
+
+            if "partition_values" in row and len(row["partition_values"]) > 0:
+                partition_values = daft.table.Table.from_pydict(
```

Comment:
You want to encode the dtype of the partition column field when creating the series here otherwise you may have a lossy cast due to type inference. See how we do it for iceberg
https://github.com/Eventual-Inc/Daft/blob/main/daft/iceberg/iceberg_scan.py#L109